File: 0775-n-ary-tree-preorder-traversal/0775-n-ary-tree-preorder-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    void recurse(Node* node, vector<int>& rec){
        if (node == nullptr) return;
        rec.push_back(node->val);
        for (auto child : node->children)
            recurse(child, rec);
    }
    
    vector<int> preorder(Node* root) {
        vector<int> result;
        recurse(root, result);
        return result;
    }
};

File: 0775-n-ary-tree-preorder-traversal/0775-n-ary-tree-preorder-traversal.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution(object):
    def preorder(self, root):
        """
        :type root: Node
        :rtype: List[int]
        """
        vec = []
        def recurse(node):
            if node is None:
                return
            vec.append(node.val)
            for child in node.children:
                recurse(child)
        recurse(root)
        return vec